open-metadata-implementation/frameworks/open-connector-framework/src/main/java/org/odpi/openmetadata/frameworks/connectors/properties/SchemaAttributes.java

```diff
@@ -30,6 +31,74 @@ public SchemaAttributes(int totalElementCount,
                             int maxCacheSize)
     {
         super(totalElementCount, maxCacheSize);
+        // Anonymous class for temporary changing the behaviour of the iterator in the case of SchemaAttributes
+        // as to not use the "totalElementCount" parameter because its value is always 0
+        super.pagingIterator = new PagingIterator(this, totalElementCount, maxCacheSize)
+        {
+            @Override
+            public ElementBase next()
+            {
+                    if (this.hasNext())
+                    {
+                        ElementBase retrievedElement = iterator.cloneElement(cachedElementList.get(cachedElementPointer));
+                        cachedElementPointer++;
+                        cachedElementStart++;
+
+                        log.debug("Returning next element:");
+                        log.debug("==> totalElementCount: " + totalElementCount);
+                        log.debug("==> cachedElementPointer: " + cachedElementPointer);
+                        log.debug("==> cachedElementStart:" + cachedElementStart);
+                        log.debug("==> maxCacheSize:" + maxCacheSize);
+
+                        return retrievedElement;
+                    }
+                    else
+                    {
+                        /*
+                         * Throw runtime exception to show the caller they are not using the list correctly.
+                         */
+                        throw new OCFRuntimeException(OCFErrorCode.NO_MORE_ELEMENTS.getMessageDefinition(this.getClass().getSimpleName()),
```

Comment:
Align parameters for getMessageDefinition under this.getClass().getSimpleName()).

Use this.getClass.getName().